clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -302,6 +335,10 @@ public String toString() {
         }
     }
 
+    public static int replicaId(FetchRequestData fetchRequestData) {
+        return fetchRequestData.replicaId() != -1 ? fetchRequestData.replicaId() : fetchRequestData.replicaState().replicaId();
```

Comment:
In the case of a consumer, we always read replica state (the second part of the statement).
I don't think there are any bugs here, but it might be good to keep in mind.